File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/Heal.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Heal : Magic
{

	// =======================
	// フィールド変数
	// =======================
	// 魔法の名前
	protected string name = "ヒール";
	// 消費MP
	protected int usemp = 20;
	// 回復量
	protected int recoverhp = 50;

	// =======================
	// コンストラクタ
	// =======================
	public Heal() : base()
	{

	}

	// =======================
	// Getter / Setter
	// =======================

	// 回復するHPの取得
	public int GetRecoverHP()
	{
		return this.recoverhp;
	}

	// =======================
	// protected メソッド
	// =======================

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================

	/**
	 * 対称プレーヤーの体力を回復する
	 * @param activePlayer {@inheritDoc}
	 * @param passivePlayer {@inheritDoc}
	 */
	public override void effect(Player activePlayer, Player passivePlayer)
	{
		// HPを 50 回復する

		activePlayer.UseMP(this.usemp);

		LogText.AddLog(string.Format("{0} の {1}！", activePlayer.GetName(), this.name));
		LogText.AddLog(string.Format("{0} の HP が  {1}　回復した！", passivePlayer.GetName(), this.recoverhp));

		passivePlayer.RecoverHP(this.recoverhp);


	}

}

File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20200716_138666938/namebattler/src/namebattler/magic/Heal.java
package namebattler.magic;

import namebattler.player.Player;

public class Heal extends Magic{
	// =======================
	// フィールド変数
	// =======================


	// =======================
	// コンストラクタ
	// =======================
	//コンストラクタ
	public Heal() {
		super();
		//名前をセット
		this.name = "ヒール";
		//使用MPをセット
		this.cost = 20;
	}

	// =======================
	// protected メソッド
	// =======================
	//個別効果
	@Override
	protected void Magieffect(Player target) {
		//ターゲットを回復
		System.out.println(target.GetName() + "を" + this.recovery + "回復！");

	}
}
